### PR TITLE
feat: upload chosen profile picture to kalium

### PR DESCRIPTION
**Pull request by trOnk12**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

### Testing

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/ui/userprofile/image/ImagePicker.kt [hunk 1]
```diff
@@ -24,22 +25,46 @@ import androidx.compose.ui.res.stringResource
...
 
 
 @OptIn(ExperimentalMaterialApi::class)
 @Composable
-fun ProfileImageScreen(avatarUrl: String) {
-    val profileImageState = rememberProfileImageState({}, {}, {})
+//TODO: the input data for ProfileImageScreen will be decided later on after sync with Yamil
+fun ImagePicker(
+    initialBitmap: Bitmap,
+    OnCloseClick: () -> Unit,
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/userprofile/image/ImagePicker.kt:44]: I think there is a typo on parameter`onCloseClick`

File: app/src/main/kotlin/com/wire/android/ui/common/bottomsheet/MenuBottomSheetItem.kt [hunk 2]
```diff
@@ -47,15 +47,11 @@ fun MenuBottomSheetItem(
     }
 }
 
-// Add Divider() at the beginning of the list, then keep adding at the end of each items, till last item
 @Composable
 fun buildMenuSheetItems(items: List<@Composable () -> Unit>) {
-    Divider(thickness = 0.5.dp)
-    items.forEachIndexed { index: Int, itemBuilder: @Composable () -> Unit ->
+    items.forEach { itemBuilder ->
+        Divider(thickness = 0.5.dp)
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/common/bottomsheet/MenuBottomSheetItem.kt:53]: I think we could use the newly created `dividerThickness` attribute on wire dimens

File: app/src/main/kotlin/com/wire/android/ui/common/imagepreview/PicturePickerFlow.kt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    val takePictureFLow = rememberTakePictureFlow({ nullableBitmap ->
+        nullableBitmap?.let {
+            pickedPicture = BitmapState.BitmapPicked(it)
+        }
+    }, { })
+
+    val openGalleryFlow = rememberOpenGalleryFlow({
+        val bitmap: Bitmap = if (Build.VERSION.SDK_INT >= 29) {
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/common/imagepreview/PicturePickerFlow.kt:47]: I wonder if this logic could be extracted to some kind of wrapper to abstract obtaining a Bitmap

**Review by yamilmedina — APPROVED** [review]
Approved! With some minor comments to be addressed, also thanks for breaking the PR into another one smaller, appreciated 😁

File: app/src/main/kotlin/com/wire/android/ui/common/imagepreview/PicturePickerFlow.kt (hunk 3 above)
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/common/imagepreview/PicturePickerFlow.kt:47]: Good idea will add TODO and address it later

File: app/src/main/kotlin/com/wire/android/ui/common/bottomsheet/MenuBottomSheetItem.kt (hunk 2 above)
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/common/bottomsheet/MenuBottomSheetItem.kt:53]: Good catch :p

**Pull request merged**